tests/lib/acpi/src/main.c

```diff
@@ -17,14 +17,43 @@ ZTEST(acpi, test_mcfg_table)
 	zassert_not_null(mcfg, "Failed to get MCFG table");
 }
 
-ZTEST(acpi, test_irq_routing_table)
+#if ACPI_DT_HAS_HID(DT_NODELABEL(pcie0))
+ZTEST(acpi, test_dev_enum)
 {
-	static ACPI_PCI_ROUTING_TABLE irq_prt_table[CONFIG_ACPI_MAX_PRT_ENTRY];
-	int status;
+	struct acpi_dev *dev;
+	ACPI_RESOURCE *res_lst;
+	int ret;
 
-	status = acpi_get_irq_routing_table(CONFIG_ACPI_PRT_BUS_NAME,
-					    irq_prt_table, ARRAY_SIZE(irq_prt_table));
-	zassert_ok(status, "Failed to get PRT");
+	dev = acpi_device_get(ACPI_DT_HID(DT_NODELABEL(pcie0)), ACPI_DT_UID(DT_NODELABEL(pcie0)));
+
+	zassert_not_null(dev, "Failed to get acpi device with given HID");
+
+	ret = acpi_current_resource_get(dev->path, &res_lst);
+
+	zassert_ok(ret, "Failed to get current resource setting");
+}
+#endif
+
+#if ACPI_DT_HAS_HID(DT_NODELABEL(rtc))
+ZTEST(acpi, test_resource_enum)
```

Comment:
I don't see you adding HID properties to any "rtc" node instances in this pull request, which makes this test essentially dead code. Please include those DTS changes if you want to add such a test case. Also, see my earlier note about using `ztest_test_skip()` for skipping tests on architectures/platforms that cannot support a certain test case.

Reply:
> I don't see you adding HID properties to any "rtc" node instances in this pull request, which makes this test essentially dead code. Please include those DTS changes if you want to add such a test case. Also, see my earlier note about using `ztest_test_skip()` for skipping tests on architectures/platforms that cannot support a certain test case.

in order to add rtc node, rtc driver need to update and I have another PR for same (https://github.com/zephyrproject-rtos/zephyr/pull/62993)

Reply:
> > I don't see you adding HID properties to any "rtc" node instances in this pull request, which makes this test essentially dead code. Please include those DTS changes if you want to add such a test case. Also, see my earlier note about using `ztest_test_skip()` for skipping tests on architectures/platforms that cannot support a certain test case.
> 
> in order to add rtc node, rtc driver need to update and I have another PR for same (#62993)

We should be able to see that the new test case passes in CI (i.e. with qemu), so you need to provide some solution already in this PR. The test case itself is not really related to rtc - all it needs is "some" device with an acpi-hid property. That's why I've been recommending to use a board-specific devicetree overlay for the test together with aliases.

If the board already has some DT node with the acpi-hid property, then you just need aliases. If it doesn't you could either add a new node or enhance an existing one with the property, and then combined that with aliases. So something like:
`tests/lib/acpi/boards/qemu_x86_64.overlay`:
```
aliases {
	acpi_dev = &some_existing_dev;
};

&some_existing_dev {
	acpi-hid = "PNP...";
};
```

And then in the test case you'd use `DT_ALIAS(acpi_dev)` instead of `DT_NODELABEL(rtc)`

Reply:
Hmm, I realise one issue with the above is that the DT binding for the device node in question needs to support the `acpi-hid` property, so you'll either need to modify some existing binding to include `acpi-device.yaml` or then you need to modify the binding in this PR.

Reply:
Btw, since pcie0 already has the acpi-hid property, couldn't you use that? In that case you'd just need to have the `aliases {` section as described above and use `&pci0` for `&some_existing_dev`. This could then later be updated to be some peripheral (once the peripheral DT bindings and drivers have been updated).